include/openthread/message.h

```diff
@@ -69,6 +69,19 @@ typedef enum otMessagePriority
     OT_MESSAGE_PRIORITY_HIGH   = 2, ///< High priority level.
 } otMessagePriority;
 
+/**
+ * Defines the OpenThread message origins.
+ *
+ */
```

Comment:
One thought here: What do we expect the behavior regarding passing back a message to host that has `OT_MESSAGE_ORIGIN_HOST_UNTRUSTED` as its origin. Should we allow it or disallow it? 
(I guess can happen if destination of message is not valid or reachable?)

Reply:
Thinking more on this: How about we define different properties/flags that can be set on an `otMessage` to control how we treat it:
- We can add a `otMessageSetAllowLookbackToHost()` that sets a `bool` flag indicating whether or not we should allow the message to be sent (back) to host.
- And independently have `otMessageSetOrigin()` but now `Origin` is limited to 3 options: 
   - `kOriginThreadNetif` 
   - `kOriginHostTrusted`
   - `kOriginHostUntrusted`

This way caller has full control of behavior they want.

Thoughts?

Reply:
Adding the `otMessageSetAllowLookbackToHost` API makes sense to me. Whether allowing loopback to host should be an independent property regarding the origin.

I'll update the PR to add this API.

Reply:
Thanks. :)